internal/datacoord/compaction.go

```diff
@@ -592,49 +622,51 @@ func (c *compactionPlanHandler) createCompactTask(t *datapb.CompactionTask) (Com
 	return task, nil
 }
 
-func (c *compactionPlanHandler) assignNodeIDs(tasks []CompactionTask) {
-	slots := c.cluster.QuerySlots()
+func assignNodeID(slots map[int64]int64, t CompactionTask) int64 {
 	if len(slots) == 0 {
-		return
+		return NullNodeID
 	}
 
-	for _, t := range tasks {
-		nodeID, useSlot := c.pickAnyNode(slots, t)
-		if nodeID == NullNodeID {
-			log.Info("compactionHandler cannot find datanode for compaction task",
-				zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("type", t.GetTaskProto().GetType().String()), zap.String("vchannel", t.GetTaskProto().GetChannel()))
-			continue
-		}
-		err := t.SetNodeID(nodeID)
-		if err != nil {
-			log.Info("compactionHandler assignNodeID failed",
-				zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("vchannel", t.GetTaskProto().GetChannel()), zap.Error(err))
-		} else {
-			// update the input nodeSlots
-			slots[nodeID] = slots[nodeID] - useSlot
-			log.Info("compactionHandler assignNodeID success",
-				zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("vchannel", t.GetTaskProto().GetChannel()), zap.Any("nodeID", nodeID))
-			metrics.DataCoordCompactionTaskNum.WithLabelValues(fmt.Sprintf("%d", NullNodeID), t.GetTaskProto().GetType().String(), metrics.Executing).Dec()
-			metrics.DataCoordCompactionTaskNum.WithLabelValues(fmt.Sprintf("%d", t.GetTaskProto().GetNodeID()), t.GetTaskProto().GetType().String(), metrics.Executing).Inc()
-		}
+	nodeID, useSlot := pickAnyNode(slots, t)
+	if nodeID == NullNodeID {
+		log.Info("compactionHandler cannot find datanode for compaction task",
+			zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("type", t.GetTaskProto().GetType().String()), zap.String("vchannel", t.GetTaskProto().GetChannel()))
+		return NullNodeID
 	}
+	err := t.SetNodeID(nodeID)
+	if err != nil {
+		log.Info("compactionHandler assignNodeID failed",
```

Comment:
Please change log level to Warning

Reply:
This is copied from previous impl, will change it later.